x-pack/platform/plugins/shared/ml/public/application/components/job_selector/group_or_job_selector_menu/get_options_for_job_selector_menu.tsx

```diff
@@ -0,0 +1,150 @@
+  page: MlPages;
+  onRemoveJobId: (jobId: string) => void;
+  removeJobIdDisabled: boolean;
+  showRemoveJobId: boolean;
+  closePopover: () => void;
+  globalState: Record<string, any>;
+  setActiveFlyout: (flyout: FlyoutType | null) => void;
+  setActiveJobId: (jobId: string | null) => void;
+  navigateToUrl: (url: string) => void;
+  share: SharePluginStart;
+}) => {
+  const pageToNavigateTo =
+    page === ML_PAGES.SINGLE_METRIC_VIEWER ? ANOMALY_EXPLORER_TITLE : SINGLE_METRIC_VIEWER_TITLE;
+  const viewInMenu: EuiContextMenuPanelItemDescriptor = {
+    name: i18n.translate(
+      'xpack.ml.overview.anomalyDetection.jobContextMenu.viewInSingleMetricViewer',
+      {
+        defaultMessage: 'View in {page}',
+        values: {
+          page: pageToNavigateTo,
+        },
+      }
+    ),
+    icon: 'visLine',
+    onClick: async () => {
+      const mlLocator = share.url.locators.get(ML_APP_LOCATOR);
+      if (!mlLocator) {
+        return;
+      }
+      const singleMetricViewerLink = await mlLocator.getUrl(
+        {
+          page:
+            page === ML_PAGES.SINGLE_METRIC_VIEWER
+              ? ML_PAGES.ANOMALY_EXPLORER
+              : ML_PAGES.SINGLE_METRIC_VIEWER,
+          pageState: {
+            globalState,
+            refreshInterval: {
+              display: 'Off',
+              pause: true,
+              value: 0,
+            },
+            jobIds: [jobId],
+            query: {
+              query_string: {
+                analyze_wildcard: true,
+                query: '*',
+              },
+            },
+          },
+        },
+        { absolute: true }
+      );
+      navigateToUrl(singleMetricViewerLink);
+
+      closePopover();
+    },
+  };
+
+  const items = [
+    {
+      name: i18n.translate('xpack.ml.overview.anomalyDetection.jobContextMenu.jobDetails', {
+        defaultMessage: 'Job details',
+      }),
+      icon: 'editorTable',
+      onClick: () => {
+        setActiveJobId(jobId);
+        setActiveFlyout(FlyoutType.JOB_DETAILS);
+        closePopover();
+      },
+    },
+    ...(showRemoveJobId
+      ? [
+          {
+            name: i18n.translate('xpack.ml.overview.anomalyDetection.jobContextMenu.removeJob', {
+              defaultMessage: 'Remove from {page}',
+              values: {
+                page:
+                  page === ML_PAGES.ANOMALY_EXPLORER
+                    ? ANOMALY_EXPLORER_TITLE
+                    : SINGLE_METRIC_VIEWER_TITLE,
+              },
+            }),
+            disabled: removeJobIdDisabled,
+            icon: 'minusInCircle',
+            onClick: () => {
+              if (onRemoveJobId) {
+                onRemoveJobId(jobId);
+                setActiveJobId(jobId);
+                setActiveFlyout(null);
+              }
+              closePopover();
+            },
+          },
+        ]
+      : []),
+    {
+      isSeparator: true,
+      key: 'separator2',
+    } as EuiContextMenuPanelItemDescriptor,
+    viewInMenu,
```

Comment:
This needs to be hidden for jobs that aren't viewable in the Single Metric Viewer.